core/src/main/java/smile/classification/OneVersusRest.java

```diff
@@ -128,12 +137,58 @@ public OneVersusRest(Classifier<T>[] classifiers, PlattScaling[] platts, IntSet
         return new OneVersusRest<>(classifiers, platts);
     }
 
+    /**
```

Comment:
It is better not to duplicate the code for Tuple/DataFrame. I would rather to check if the classifier support `f(x)` function (by catching the exception). If not, don't fit platt scaling.

Reply:
I agree there is a lot of duplicated code and I do not like it either. But the main reason I defined a new fit method is because the types do not match with the previous fit method: the previous fit takes parameters (T[] x, int[] y, int pos, int neg, BiFunction<T[], int[], Classifier<T>> trainer) and there is no simple way to apply it on a DataFrame that is why I defined a new fit method with parameters (Formula formula, DataFrame data, int pos, int neg, BiFunction<Formula, DataFrame, SoftClassifier<Tuple>> trainer).

But if we simply add a check for support for f function, then the workaround I proposed in the issue #607 should work:
```
val xt = x.map(arr => Tuple.of(arr, zip.drop(0).schema()))
val model = ovr(xt, y) { (xt, y) => RandomForest.fit("class" ~, DataFrame.of(xt.map(_.toArray)).merge(DataFrame.of(y.map(Array(_)), "class")))}
```
(Indeed it works, but I did not test the accuracy because cv fails:
```
cv.classification(10, xt, y) { case (xt, y) => ovr(xt, y) { (xt, y) => RandomForest.fit("class" ~, DataFrame.of(xt.map(_.toArray)).merge(DataFrame.of(y.map(Array(_)), "class"))) } }
```
fails with the error:
```
<console>:49: error: value toArray is not a member of (Double, Double, Double, Double, Double, Double, Double, Double, Double, Double, Double, Double, Double, Double, Double, Double, Double, Double, Double, Double, Double, Double)
       cv.classification(10, xt, y) { case (xt, y) => ovr(xt, y) { (xt, y) => RandomForest.fit("class" ~, DataFrame.of(xt.map(_.toArray)).merge(DataFrame.of(y.map(Array(_)), "class"))) } }
```
)
But I am worried about the efficiency of converting array of tuples  x to array of arrays of doubles, to dataframe and merging with dataframe of labels converted from array of ints y, and all of this k times (k being the number of classes).